src/site/_filters/paged-navigation.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Takes an object and determines how many links to other pages in the same paginated collection to display.
+ * @param {object} paged Page details including the `index` in the pagination, how many `pages` there are, and the `href` of the page.
+ * @return {Array<object>} An array of up to 8 items to display, including href and index.
+ */
+module.exports = function pagedNavigation(paged) {
+  const shiftBy = 4;
+  const start = paged.index - shiftBy > 0 ? paged.index - shiftBy : 0;
+  const end =
+    paged.index + shiftBy < paged.pages ? paged.index + shiftBy : paged.pages;
+
+  const pagesToShow = Array.from({
+    length: end - start,
+  }).map((_, i) => {
+    const index = i + start + 1;
+    return {
+      href: index === 1 ? paged.href : paged.href + "/" + index,
+      index,
+    };
+  });
+
+  const lastPageToShow = pagesToShow[pagesToShow.length - 1];
+
+  if (lastPageToShow.index !== paged.pages) {
```

Comment:
check if `lastPageToShow` is truthy, as this would crash for an empty collection?

Reply:
Should never be an issue, but will do.